test/common/http/utility_test.cc

```diff
@@ -356,6 +356,41 @@ initial_connection_window_size: 65535
   }
 }
 
+// Verify that values are pulled from the defaults profile.
+TEST(HttpUtility, TestDefaultsProfile) {
+  {
+    std::unique_ptr<ScopedDefaultsProfileSingleton> dp =
+        std::make_unique<ScopedDefaultsProfileSingleton>(
+            std::make_unique<DefaultsProfile>(DefaultsProfile::Safe));
+    DefaultsProfile::ConfigContext context = DefaultsProfile::ConfigContext(
+        "envoy.extensions.filters.network.http_connection_manager.v3."
+        "HttpConnectionManager.http2_protocol_options");
+
+    envoy::config::core::v3::Http2ProtocolOptions http2_options;
+    http2_options = ::Envoy::Http2::Utility::initializeAndValidateOptions(context, http2_options);
+
+    EXPECT_EQ(100, http2_options.max_concurrent_streams().value());
+    EXPECT_EQ(65536, http2_options.initial_stream_window_size().value());
+    EXPECT_EQ(1048576, http2_options.initial_connection_window_size().value());
+  }
+
+  {
+    // invoke DefaultsProfile constructor with no args, creating a blank profile
+    std::unique_ptr<ScopedDefaultsProfileSingleton> dp =
```

Comment:
nit: typically when std::make_unique is used the type of the variable is just auto, as it is known what type make_unique creates.